### DISCUSSION TITLE
Codecov uploading

### DISCUSSION THREAD

**Discussion by lcnbr** [ipetkov/crane#636] [Q&A] [answered]
I am trying to set up codecoverage in CI using tarpaulin but I cannot figure out how to tell the Codecov action where to look for the coverage output files. Does anyone have experience with this?

**Comment by dpc** [comment]
https://github.com/fedimint/fedimint/blob/ea82bdb2980db6c4d2aa1aefd9a4cef92e8147f4/.github/workflows/ci-nix.yml#L290

**Reply by lcnbr** [reply]
Ah but where do you set the output directory to be results?

**Reply by dpc** [reply, marked as answer]
`result` is a standard name for a symlink that `nix run` makes after successful run.

From inside of the derivation you use normal `$out` https://github.com/fedimint/fedimint/blob/ea82bdb2980db6c4d2aa1aefd9a4cef92e8147f4/nix/flakebox.nix#L406C1-L406C2

**Reply by ipetkov** [reply]
Worth noting that if you can also control the name of the symlink nix will make, e.g. `nix build .#myTarpaulinDerivation -o mycov` and then the result will be in `mycov/cobertura.xml`

**Reply by lcnbr** [reply]
Ah but I was using flake check directly, so I cannot set that. I guess I can just use build!

**Reply by lcnbr** [reply]
It only works with build I think. And the output is always in the result dir, even when the `$out` variable is set to a different directory...

**Reply by dpc** [reply]
@lcnbr It will only work with `build`, but you can run check first, and then `nix build` to get the symlink. 

`$out` is an input to your derrivation and you can't change it. The `-o` to `nix build` controls the name of the symlink to the atuall build in the nix store.
